libraries/resource_provider.rb

```
@@ -183,6 +183,46 @@ def create(method = :create)
       end
     end
 
+    # Adds Oneview Scope to the Oneview resource if scope is not already added
+    def add_scope
+      @item.retrieve!
+      scope = load_resource(:Scope, @context.scope)
```

What if the scope is nil?

---

Really, I should throw a error if scope is not present. I'll do it.